drv/ignition-server/src/main.rs

```diff
@@ -187,47 +301,55 @@ impl idl::InOrderIgnitionImpl for ServerImpl {
             return Err(RequestError::from(IgnitionError::InvalidPort));
         }
 
-        let port_state = self.port_state(port).map_err(RequestError::from)?;
-        let target_state = port_state.target().ok_or_else(|| {
-            RequestError::from(IgnitionError::NoTargetPresent)
-        })?;
+        ringbuf_entry!(Trace::SystemPowerRequest(port, request));
 
-        if target_state.system_power_reset_in_progress()
-            || target_state.system_power_off_in_progress()
-            || target_state.system_power_on_in_progress()
-        {
-            return Err(RequestError::from(IgnitionError::RequestInProgress));
+        match self.target_request(port, request) {
```

Comment:
Since we leave the `Ok` option unchanged, this could be something like
```rust
map_err(|e| {
    ringbuf_entry!(Trace::SystemPowerRequestError(port, e));
    Err(RequestError::from(e))
});
```